lib/models/LifecycleConfiguration.js

```diff
@@ -0,0 +1,623 @@
+                    'Rule includes more than one Prefix');
+                return filterObj;
+            }
+            filterObj.rulePrefix = filter[0];
+            return filterObj;
+        }
+        if (filter.And && (filter.Prefix || filter.Tag) ||
+        (filter.Prefix && filter.Tag)) {
+            filterObj.error = errors.MalformedXML.customizeDescription(
+                'Filter should only include one of And, Prefix, or Tag key');
+            return filterObj;
+        }
+        if (filter.Prefix) {
+            if (filter.Prefix.length > 1) {
+                filterObj.error = errors.MalformedXML.customizeDescription(
+                    'Filter includes more than one Prefix');
+                return filterObj;
+            }
+            filterObj.rulePrefix = filter.Prefix[0];
+            return filterObj;
+        }
+        if (filter.Tag) {
+            filterObj.tags = [];
+            if (this._checkTagError(filter.Tag[0])) {
+                filterObj.error = errors.MalformedXML.customizeDescription(
+                    'Tag XML is invalid');
+                return filterObj;
+            }
+            filterObj.tags.push({
+                key: filter.Tag[0].Key[0],
+                val: filter.Tag[0].Value[0],
+            });
+            return filterObj;
+        }
+        if (filter.And) {
+            const andF = filter.And[0];
+            filterObj.tags = [];
+            if (!andF.Tag || (!andF.Prefix && andF.Tag.length < 2)) {
+                filterObj.error = errors.MalformedXML.customizeDescription(
+                    'And should include Prefix and Tag or more than one Tag');
+                return filterObj;
+            }
+            if (andF.Prefix && andF.Prefix.length > 1) {
+                filterObj.error = errors.MalformedXML.customizeDescription(
+                    'And includes more than one Prefix');
+                return filterObj;
+            }
+            if (this._checkTagError(andF.Tag[0])) {
+                filterObj.error = errors.MalformedXML.customizeDescription(
+                    'Tags XML is invalid');
+                return filterObj;
+            }
+            filterObj.rulePrefix = andF.Prefix ? andF.Prefix[0] : null;
+            for (let i = 0; i < andF.Tag[0].Key.length; i++) {
+                const tag = {
+                    key: andF.Tag[0].Key[i],
+                    val: andF.Tag[0].Value[i],
+                };
+                filterObj.tags.push(tag);
+            }
+            return filterObj;
+        }
+        return filterObj;
+    }
+
+    /**
+     * Check that each tag object is valid
+     * @param {object} tag - a tag object from a filter object
+     * @return {boolean} - indicates whether tag object is valid
+     */
+    _checkTagError(tag) {
+        if (!tag.Key || !tag.Value) {
+            return true;
+        }
+        if (tag.Key.length !== tag.Value.length) {
```

Comment:
also checking here, why must the tag key and value be the same length?

Reply:
the Key and Value array lengths should be the same because there has to be one value for every key. wouldn't do to have 5 keys and only 4 values, or the other way around :)